numpy/core/src/multiarray/conversion_utils.c

```diff
@@ -110,7 +145,43 @@ PyArray_IntpConverter(PyObject *obj, PyArray_Dims *seq)
         return NPY_SUCCEED;
     }
 
-    len = PySequence_Size(obj);
+    /*
+    * If obj is a scalar we skip all the useless computations and jump to
+    * intp_from_scalar as soon as possible.
+    */
+    if (PyLong_CheckExact(obj) || !PySequence_Check(obj)) {
+        len = 1;
+
+        seq->ptr = npy_alloc_cache_dim(1);
+        if (seq->ptr == NULL) {
+            PyErr_NoMemory();
+            return NPY_FAIL;
+        }
+        seq->len = len;
+
+        npy_intp rvalue = intp_from_scalar(obj, (npy_intp *)seq->ptr, 0);
```

Comment:
Could you rewrite this so that it reads:
```
seq->ptr[0] = intp_from_scalar(obj);
if (error_converting(seq->ptr[0]) {
    npy_free_cache_dim_obj(*seq);
    seq->ptr = NULL;  /* I would think this is unnecessary, but keep it if it was there? */
    return NPY_FAIL;
}
```

Reply:
I.e. I am thinking that you should simplify `intp_from_scalar`.

Reply:
I thought for some minutes about the fact that `intp_from_scalar` was too verbose (to call, and also in its implementation), but then I decided to stick with the "format" of the other functions.

Fixed in 8dc0f00